centreon/www/include/configuration/configObject/host/DB-Func.php

```diff
@@ -1738,54 +1790,65 @@ function updateHost_MC($host_id = null)
     }
 
     if (isset($ret['criticality_id']) && $ret['criticality_id']) {
-        setHostCriticality($host_id, $ret['criticality_id']);
+        setHostCriticality($hostId, $ret['criticality_id']);
     }
 
-    //@TODO: Check if Macro value are passwords
-
     $vaultConfiguration = getVaultConfiguration();
+
     //If there is a vault configuration write into vault
     if ($vaultConfiguration !== null) {
         try {
-            $passwordTypeData = addHostSNMPCommunityToPasswordType($bindParams);
             $httpClient = new CentreonRestHttp();
             $logger = getLogger();
             $clientToken = authenticateToVault($vaultConfiguration, $logger, $httpClient);
-            $hostSecrets = getHostSecretsFromVault(
+            $hostSecretsFromVault = getHostSecretsFromVault(
                 $vaultConfiguration,
-                (int) $host_id,
+                (int) $hostId,
                 $clientToken,
                 $logger,
                 $httpClient
             );
+            $macros = $hostObj->getFormattedMacros();
+            $macroPasswordIds = getIdOfUpdatedPasswordMacros($macros);
+            $updateHostPayload = prepareUpdateMCPayload(
+                $bindParams[':host_snmp_community'][\PDO::PARAM_STR],
+                $macros,
+                $hostSecretsFromVault
+            );
 
-            if (! empty($passwordTypeData)) {
-                //Replace olds vault values by the new ones
-                foreach($passwordTypeData as $keyName => $value) {
-                    $hostSecrets[$keyName] = $value;
-                }
+            // If no more fields are password types, we delete the host from the vault has it will not be read.
+            if (empty($updateHostPayload)) {
+                deleteHostFromVault($vaultConfiguration, (int) $hostId, $clientToken, $logger, $httpClient);
+            } else {
                 writeSecretsInVault(
                     $vaultConfiguration,
-                    $host_id,
+                    $hostId,
                     $clientToken,
-                    $hostSecrets,
+                    $updateHostPayload,
                     $logger,
                     $httpClient
                 );
-
-                updateHostTablesWithVaultPath($vaultConfiguration, $host_id, $pearDB);
+                $hostPath = "secret::" . $vaultConfiguration->getId() . "::"
+                    . $vaultConfiguration->getStorage()
+                    . "/monitoring/hosts/" . $hostId;
+                if (array_key_exists(SNMP_COMMUNITY_MACRO_NAME, $updateHostPayload)){
+                    updateHostTableWithVaultPath($pearDB, $hostPath, $hostId);
+                }
+                if (! empty($macroPasswordIds)) {
+                    updateOnDemandMacroHostTableWithVaultPath($pearDB, $macroPasswordIds, $hostPath);
+                }
```

Comment:
```suggestion
                if (array_key_exists(SNMP_COMMUNITY_MACRO_NAME, $updateHostPayload)){
                    updateHostTableWithVaultPath($pearDB, $hostPath, $hostId);
                }
                if (! empty($macroPasswordIds)) {
                    updateOnDemandMacroHostTableWithVaultPath($pearDB, $macroPasswordIds, $hostPath);
                }
```
Add comment on table writtten